load-test-framework/src/main/java/com/google/pubsub/clients/common/LoadTestRunner.java

```diff
@@ -94,33 +121,8 @@ public void run() {
         }
       }
     });
-
-    StartRequest request = requestFuture.get();
-    log.info("Request received, starting up server.");
-    ListeningExecutorService executor = MoreExecutors.listeningDecorator(
-        Executors.newFixedThreadPool(request.getMaxOutstandingRequests() + 10));
-
-    final RateLimiter rateLimiter = RateLimiter.create(request.getRequestRate());
-    final Semaphore outstandingTestLimiter =
-        new Semaphore(request.getMaxOutstandingRequests(), false);
-    client = loadFunction.apply(request);
-
-    final long toSleep = request.getStartTime().getSeconds() * 1000 - System.currentTimeMillis();
-    if (toSleep > 0) {
-      Thread.sleep(toSleep);
-    }
-
-    stopwatch.start();
-    while (shouldContinue(request)) {
-      outstandingTestLimiter.acquireUninterruptibly();
-      rateLimiter.acquire();
-      executor.submit(client).addListener(outstandingTestLimiter::release, executor);
-    }
-    stopwatch.stop();
-    executor.shutdownNow();
-    finished.set(true);
-    finishedFuture.get();
-    log.info("Load test complete, shutting down.");
+    // Deadlock forever, since we do not want the server to stop.
+    Thread.currentThread().join();
```

Comment:
Now that we don't want to stop after a single run, we need to have a thread sit doing nothing while the work is scheduled. Before we just waited on isFinished, but that doesn't make sense anymore since we'll run multiple tests.